File: N2775.java
import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;

public class N2775 {
	public static void main(String[] args) throws IOException {
		
		BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
		
		int TestCase = Integer.parseInt(br.readLine());
		
		for (int i=0; i<TestCase; i++)
		{
			int k = Integer.parseInt(br.readLine());
			int n = Integer.parseInt(br.readLine());
			System.out.println(Combination((k+n), (k+1)));
		}
	}
	
	public static int Combination(int n, int r)
	{
		int result = 1;
		
		for(int i=1; i<=r; i++)
		{ result *= (n-i+1); result/=i; }
		return result;
	}
}

File: N2775.kt
import java.io.*;
import java.util.*;

private val bw = BufferedWriter(OutputStreamWriter(System.`out`))

var apt = Array(15, { IntArray(15, { i -> 0 }) })
var k: Int = 0
var n: Int = 0

fun main(args: Array<String>) {
    buildApt()
    init()
    solution()
}

fun buildApt() {
    for (x in 1..14) {
        apt[0][x] = x
    }

    for (x in 1..14) {
        for (y in 1..14) {
            apt[x][y] = apt[x - 1][y] + apt[x][y - 1]
        }
    }
}

private fun init() {
    val br = BufferedReader(InputStreamReader(System.`in`))
    var tc: Int = br.readLine().toInt()

    for (x in 1..tc) {
        k = br.readLine().toInt()
        n = br.readLine().toInt()
        bw.write(apt[k][n].toString()+"\n")
    }
}

private fun solution() {
    bw.flush()
    bw.close()
}
